lib/countly.js

```diff
@@ -43,221 +43,153 @@
+            }
+            else if (!this.device_id) {
+                this.device_id = "[CLY]_temp_id";
+            }
+            this.url = stripTrailingSlash(getConfig("url", ob, ""));
+            this.app_version = getConfig("app_version", ob, "0.0");
+            this.country_code = getConfig("country_code", ob, null);
+            this.city = getConfig("city", ob, null);
+            this.ip_address = getConfig("ip_address", ob, null);
+            this.ignore_bots = getConfig("ignore_bots", ob, true);
+            this.force_post = getConfig("force_post", ob, false);
+            this.remote_config = getConfig("remote_config", ob, false);
+            this.ignore_visitor = getConfig("ignore_visitor", ob, false);
+            this.require_consent = getConfig("require_consent", ob, false);
+            if (!Array.isArray( ignoreReferrers ) ) {
+                ignoreReferrers = [];
             }
-
-            if (Countly.url === "") {
+    
+            if (this.url === "") {
                 log("Please provide server URL");
-                Countly.ignore_visitor = true;
+                this.ignore_visitor = true;
             }
             if (store("cly_ignore")) {
                 //opted out user
-                Countly.ignore_visitor = true;
+                this.ignore_visitor = true;
             }
             
+            requestQueue = store("cly_queue") || [];
+            eventQueue = store("cly_event") || [];
+            remoteConfigs = store("cly_remote_configs") || {};
+            
             checkIgnore();
             
             if (window.name && window.name.indexOf("cly:") === 0) {
                 try{
-                    Countly.passed_data = JSON.parse(window.name.replace("cly:", ""));
+                    this.passed_data = JSON.parse(window.name.replace("cly:", ""));
                 } catch(ex){log("Could not parse name", window.name);}
             }
             else if (location.hash && location.hash.indexOf("#cly:") === 0) {
                 try{
-                    Countly.passed_data = JSON.parse(location.hash.replace("#cly:", ""));
+                    this.passed_data = JSON.parse(location.hash.replace("#cly:", ""));
                 } catch(ex){log("Could not parse hash", location.hash);}
             }
-
-            if (Countly.passed_data) {
-                if (Countly.passed_data.token && Countly.passed_data.purpose) {
-                    if (Countly.passed_data.token !== store("cly_old_token")) {
-                        setToken(Countly.passed_data.token);
-                        store("cly_old_token", Countly.passed_data.token);
-                    }
-                    if (Countly.passed_data.purpose === "heatmap") {
-                        Countly.ignore_visitor = true;
+    
+            if ((this.passed_data && this.passed_data.app_key && this.passed_data.app_key == this.app_key) || (this.passed_data && !this.passed_data.app_key && global)) {
```

Comment:
Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/manual/ar2rsawseen/countly-sdk-web/pullRequest?prid=6356797)